### PR TITLE
Update to latest KFP SDK release (0.2.2)

### PR DESCRIPTION
**Pull request by ckadner**
Also adding more detailed setup instructions using a separate build directory for the KFP compiler files to not override the `$GOPATH/../kubeflow/pipelines/sdk`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfp-tekton/13)

### CODE REVIEW

File: sdk/python/kfp/compiler/compiler.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
 from ._default_transformers import add_pod_env
 
-from ..components._structures import InputSpec
+from ..components.structures import InputSpec
```
Comment (Tomcli) [inline sdk/python/kfp/compiler/compiler.py]: I remember all the module under components has underscore prefix. Did they changed recently?

Comment (ckadner) [reply sdk/python/kfp/compiler/compiler.py]: yes, see https://github.com/kubeflow/pipelines/commit/c83aff2738f77778802fa9fb034bcacc233d125e

**Pull request merged**